src/ast/transform/SimplifyConstantBinaryConstraints.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2018, The Souffle Developers. All rights reserved
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file SimplifyConstantBinaryConstraints.h
+ *
+ ***********************************************************************/
+
+#pragma once
+
+#include "ast/TranslationUnit.h"
+#include "ast/transform/Transformer.h"
+#include <string>
+
+namespace souffle::ast::transform {
+
+/**
+ * Eliminate clauses containing unsatisfiable constant binary constraints
```

Comment:
this transform does not eliminate clauses, it simplifies binary constraints into boolean constraints.

Reply:
Good catch, updated the docstring.